File: android/src/main/java/org/maplibre/reactnative/components/AbstractMapFeature.java
package org.maplibre.reactnative.components;

import android.content.Context;

import com.facebook.react.views.view.ReactViewGroup;
import org.maplibre.reactnative.components.mapview.MLRNMapView;

public abstract class AbstractMapFeature extends ReactViewGroup {
    public AbstractMapFeature(Context context) {
        super(context);
    }

    public abstract void addToMap(MLRNMapView mapView);
    public abstract void removeFromMap(MLRNMapView mapView);
}

File: android/src/main/java/org/maplibre/reactnative/components/AbstractMapFeature.kt
package org.maplibre.reactnative.components

import android.content.Context
import com.facebook.react.views.view.ReactViewGroup
import org.maplibre.reactnative.components.mapview.MLRNMapView

abstract class AbstractMapFeature(
    context: Context?,
) : ReactViewGroup(context) {
    abstract fun addToMap(mapView: MLRNMapView)

    abstract fun removeFromMap(mapView: MLRNMapView)
}
